photon-core/src/main/java/org/photonvision/vision/pipeline/DnnPipeline.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (C) Photon Vision.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.photonvision.vision.pipeline;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.opencv.core.Core;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfFloat;
+import org.opencv.core.MatOfInt;
+import org.opencv.core.MatOfRect2d;
+import org.opencv.core.Point;
+import org.opencv.core.Rect2d;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.dnn.Dnn;
+import org.opencv.dnn.Net;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.utils.Converters;
+import org.photonvision.common.util.ColorHelper;
+import org.photonvision.vision.frame.Frame;
+import org.photonvision.vision.frame.FrameThresholdType;
+import org.photonvision.vision.pipe.impl.*;
+import org.photonvision.vision.pipeline.result.CVPipelineResult;
+import org.photonvision.vision.target.TrackedTarget;
+
+public class DnnPipeline extends CVPipeline<CVPipelineResult, DnnPipelineSettings> {
+    private final CalculateFPSPipe calculateFPSPipe = new CalculateFPSPipe();
+
+    private static final FrameThresholdType PROCESSING_TYPE = FrameThresholdType.NONE;
+
+    public DnnPipeline() {
+        super(PROCESSING_TYPE);
+        settings = new DnnPipelineSettings();
+    }
+
+    Net net = null;
+    private List<String> outBlobNames = List.of();
+
+    private List<String> coco_names;
+
+    public DnnPipeline(DnnPipelineSettings settings) {
+        super(PROCESSING_TYPE);
+        this.settings = settings;
+
+        // Downloaded from https://dev.to/kojix2/yolov7-object-detection-in-ruby-in-10-minutes-5cjh
+        // https://s3.ap-northeast-2.wasabisys.com/pinto-model-zoo/307_YOLOv7/with-postprocess/resources_post.tar.gz
+        try {
+            // this.net = Dnn.readNetFromONNX("/home/matt/Downloads/best_1.onnx");
+            // this.net = Dnn.readNet("/home/matt/Downloads/yolov7_post_640x640.onnx");
+            this.net =
+                    Dnn.readNetFromDarknet(
+                            "/home/matt/Downloads/yolov4-csp-swish.cfg",
+                            "/home/matt/Downloads/yolov4-csp-swish.weights");
```

Comment:
For these, you're either going to need to make these configurable or ship them as assets in the jar. 

If you choose to go the jar route you should be able to drop the files in `photo-core/src/main/resources/` and then access them with `this.getClass().getResourceAsStream("yolov4-csp-swish.cfg"))` and `this.getClass().getResourceAsStream("yolov4-csp=swish.weights"))`

Reply:
Yeah, these are just some random Yolov4 weights someone trained on the coco dataset. For FRC stuff you'd wanna train your own off of a dataset like 900's or something I imagine?